app/controllers/choose_roles_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class ChooseRolesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_choose_role_form, only: %i[show create]
+
+  def show
+    if @choose_role_form.only_one_role || @choose_role_form.has_no_role
+      redirect_to @choose_role_form.redirect_path(helpers:)
```

Comment:
something maybe in the next AB prs: we need to watch out for how many redirects we're doing to get the single AB role page, and to avoid too many